src/canvasdrawer.js

```diff
@@ -0,0 +1,1151 @@
+                    this._restoreRotationChanges(false);
+                }
+                if (this.viewport.getRotation(true) % 360 !== 0) {
+                    this._restoreRotationChanges(false);
+                }
+            }
+        }
+
+        if (!sketchScale) {
+            if (this.viewport.getRotation(true) % 360 === 0 &&
+                tiledImage.getRotation(true) % 360 === 0) {
+                if(this.viewer.viewport.getFlip()) {
+                    this._flip();
+                }
+            }
+        }
+
+        this._drawDebugInfo( tiledImage, lastDrawn );
+
+        // Fire tiled-image-drawn event.
+        if( this.viewer ){
+            /**
+                * Raised when a tiled image is drawn to the canvas. Only valid
+                * for webgl and canvas drawers.
+                *
+                * @event tiled-image-drawn
+                * @memberof OpenSeadragon.Viewer
+                * @type {object}
+                * @property {OpenSeadragon.Viewer} eventSource - A reference to the Viewer which raised the event.
+                * @property {OpenSeadragon.TiledImage} tiledImage - Which TiledImage is being drawn.
+                * @property {Array} tiles - An array of Tile objects that were drawn.
+                * @property {?Object} userData - Arbitrary subscriber-defined object.
+                */
+            this.viewer.raiseEvent( 'tiled-image-drawn', {
+                tiledImage: tiledImage,
+                tiles: lastDrawn,
+            });
+        }
+    }
+
+    /**
+         * @private
+         * @inner
+         * This function converts the given point from to the drawer coordinate by
+         * multiplying it with the pixel density.
+         * This function does not take rotation into account, thus assuming provided
+         * point is at 0 degree.
+         * @param {OpenSeadragon.Point} point - the pixel point to convert
+         * @returns {OpenSeadragon.Point} Point in drawer coordinate system.
+         */
+    _viewportCoordToDrawerCoord(point) {
+        var vpPoint = this.viewport.pixelFromPointNoRotate(point, true);
+        return new $.Point(
+            vpPoint.x * $.pixelDensityRatio,
+            vpPoint.y * $.pixelDensityRatio
+        );
+    }
+
+    /**
+     * @private
+     * @inner
+     * Draws special debug information for a TiledImage if in debug mode.
+     * @param {OpenSeadragon.Tile[]} lastDrawn - An unordered list of Tiles drawn last frame.
+     */
+    _drawDebugInfo( tiledImage, lastDrawn ) {
+        if( tiledImage.debugMode ) {
+            for ( var i = lastDrawn.length - 1; i >= 0; i-- ) {
+                var tile = lastDrawn[ i ];
+                try {
+                    this._drawDebugInfoOnTile(tile, lastDrawn.length, i, tiledImage);
+                } catch(e) {
+                    $.console.error(e);
+                }
+            }
+        }
+    }
+
+    /* Methods from Tile */
```

Comment:
Is this just a historic reference to where these functions came from? If so, it's probably not an appropriate comment. That said, I'm all for using comments like this to define sections. Just reword it so it applies to the organization going forward, rather than the history of how things got here.

Also, probably anything that's not a doc comment should use the `//` comments.

Reply:
Yes, it was just to point out where the methods came from to make review easier. It can be deleted now.